src/optimizer/OptRule.cpp

```diff
@@ -71,7 +71,7 @@ StatusOr<MatchedResult> OptRule::match(OptContext *ctx, const OptGroupNode *grou
     return matched;
 }
 
-bool OptRule::match(OptContext *ctx, const MatchedResult &matched) const {
+bool OptRule::match(OptContext *ctx, const MatchedResult &matched) {
```

Comment:
I removed the const feature of `OptRule::match` bcz I expected the member functions within class `OptRule` could communicate through member variables. This is usually useful in certain optimization scenarios involving expression rewriting.